backend/typescript/services/implementations/camperService.ts

```diff
@@ -19,89 +18,80 @@ const Logger = logger(__filename);
 
 class CamperService implements ICamperService {
   /* eslint-disable class-methods-use-this */
-  async createCamper(camper: CreateCamperDTO): Promise<CamperDTO> {
-    let newCamper: Camper;
+  async createCampers(campers: CreateCampersDTO): Promise<Array<CamperDTO>> {
+    let newCamperDTOs: Array<CamperDTO> = [];
+    let newCampers: Array<Camper> = [];
     let existingCamp: CampSession | null;
     try {
-      newCamper = await MgCamper.create({
-        campSession: camper.campSession,
-        firstName: camper.firstName,
-        lastName: camper.lastName,
-        age: camper.age,
-        allergies: camper.allergies,
-        hasCamera: camper.hasCamera,
-        hasLaptop: camper.hasLaptop,
-        earlyDropoff: camper.earlyDropoff,
-        latePickup: camper.latePickup,
-        specialNeeds: camper.specialNeeds,
-        contacts: camper.contacts,
-        registrationDate: camper.registrationDate,
-        hasPaid: camper.hasPaid,
-        chargeId: camper.chargeId,
-        formResponses: camper.formResponses,
-        charges: camper.charges,
-      });
-
-      try {
-        existingCamp = await MgCamp.findByIdAndUpdate(
-          camper.campSession,
-          {
-            $push: { campers: newCamper.id },
-          },
-          { runValidators: true },
-        );
-
-        if (!existingCamp) {
-          throw new Error(`CampSession ${camper.campSession} not found.`);
-        }
-      } catch (mongoDbError: unknown) {
-        // rollback user creation
+      newCampers = await MgCamper.insertMany(campers);
+      if (campers.length > 0) {
         try {
-          const deletedCamper: Camper | null = await MgCamper.findByIdAndDelete(
-            newCamper.id,
+          existingCamp = await MgCampSession.findByIdAndUpdate(
+            campers[0].campSession,
+            {
+              $push: { campers: newCampers[0].id },
```

Comment:
i think we want to push all of the new campers ids here

Reply:
truee. done!